src/herbivoresAndCarnivores.js

```diff
@@ -1,19 +1,36 @@
 'use strict';
 
-class Animal {
-  // write your code here
-}
+/**
+ * @param {Object} state
+ * @param {Object[]} actions
+ *
+ * @return {Object[]}
+ */
+function transformStateWithClones(state, actions) {
+  // write code here
+  let stateCopy = { ...state };
+  const actionsList = [];
 
-class Herbivore extends Animal {
-  // write your code here
-}
+  for (const action of actions) {
+    switch (action.type) {
+      case 'addProperties':
+        Object.assign(stateCopy, action.extraData);
+        break;
+
+      case 'removeProperties':
+        for (const key of action.keysToRemove) {
+          delete stateCopy[key];
+        }
+        break;
+
+      case 'clear':
+        stateCopy = {};
+        break;
+    }
+    actionsList.push({ ...stateCopy });
+  }
 
-class Carnivore extends Animal {
-  // write your code here
+  return actionsList;
 }
 
-module.exports = {
-  Animal,
-  Herbivore,
-  Carnivore,
-};
+module.exports = transformStateWithClones;
```

Comment:
This code does not seem to relate to the task description provided. The task is to create an Animal class with Herbivore and Carnivore subclasses, but this function deals with state transformation and cloning. Please ensure that the code provided corresponds to the task requirements.